pennylane/operation.py

```diff
@@ -474,45 +390,152 @@ def parameters(self):
         return _unflatten(temp_val, self.params)[0]
 
     def queue(self):
-        """Append the operation to a QNode queue."""
+        """Append the operator to a QNode queue."""
         if QNode._current_context is None:
             raise QuantumFunctionError("Quantum operations can only be used inside a qfunc.")
 
         QNode._current_context._append_op(self)
         return self  # so pre-constructed Observable instances can be queued and returned in a single statement
```

Comment:
Yep. This was discussed briefly a couple of weeks ago, but the reason is that since observables are returned (and directly accessible to the QNode that way), they really don't need to be queued. It also makes the logic in the `measure.py` module cleaner (currently, we are removing observables from the queue, setting the return type, and then _re_queuing).

Reply:
The queuing makes it possible to detect Observable instances that are constructed in the qfunc but not returned, and raise an error.

Regarding measure.py, how about modifying the op in place in the queue? Shouldn't it always be the very latest op added?

Reply:
That's a good point